demo/config/targets/SYSTEM/lib/limits_groups.rb

```diff
@@ -11,28 +11,44 @@
 require 'cosmos/tools/cmd_tlm_server/limits_groups_background_task'
 
 module Cosmos
+  # Inheriting from the LimitsGroupsBackgroundTask provides a framework for
+  # automatically enabling and disabling limits groups based on telemetry.
   class LimitsGroups < LimitsGroupsBackgroundTask
-    def initialize(initial_delay)
-      super(initial_delay)
-      @temp1_enable_code = Proc.new do
-        enable_limits_group('GROUND')
+    def initialize(initial_delay = 0, task_delay = 0.5)
+      # Initial delay upon starting the server before staring the group checks
+      # followed by the background task delay between check iterations
+      super(initial_delay, task_delay)
+      # Creating a Proc allows for arbitrary code to be executed when a group
+      # is enabled or disabled.
+      @temp2_enable_code = Proc.new do
+        enable_limits_group('GROUND') # Enable the GROUND group
       end
-      @temp1_disable_code = Proc.new do
-        disable_limits_group('GROUND')
+      @temp2_disable_code = Proc.new do
+        disable_limits_group('GROUND') # Disable the GROUP group
       end
     end
 
-    def check_temp1
-      process_group(3, @temp1_enable_code, @temp1_disable_code) do
-        tlm("INST HEALTH_STATUS TEMP1") > 0
-      end
-    end
-
-    def check_temp234
-      process_group(5) do
-        (tlm("INST HEALTH_STATUS TEMP2") > 0) ||
-        (tlm("INST HEALTH_STATUS TEMP3") > 0) ||
-        (tlm("INST HEALTH_STATUS TEMP4") > 0)
+    # This method name is important. It must begin with 'check_' and end with
+    # the name of a Limits Group defined by the LIMITS_GROUP keyword. In the
+    # demo this is defined in config/targets/SYSTEM/cmd_tlm/limits_groups.txt.
+    # Note that there is a LIMITS_GROUP TEMP2 so we have a match.
+    def check_temp2
+      # Inside this method you must call process_group. The first parameter is
+      # the number of seconds to delay before enabling the group when the telemetry
+      # check is true. When the telemetry check is false the group is instantly
+      # disabled. The next two parameters are Proc objects which are called with
+      # the group is enabled and disabled respectively. We defined our Proc objects
+      # in the constructor to enable and disable the GROUND group.
+      process_group(0, @temp2_enable_code, @temp2_disable_code) do
+        val = tlm("INST HEALTH_STATUS TEMP2")
```

Comment:
Can we move this to INST2 instead of INST, so that it is less obvious and weird to see the limits being disabled?

Reply:
Ha, I read INST2 as TEMP2. My bad. I'll move it.